### ISSUE TITLE
Capacitor banks misbehaving with tesserack after relog

### ISSUE THREAD

**Issue by artubi** [SleepyTrousers/EnderIO#2979]
Hi,
I have found a really odd issue. Capacitor banks behaves oddly after relogging into map. I have done some tests and it seem do only if TE Tesserack is connected into it. Even via conduit.

Tested with:
EnderIO-1.7.10-2.3.0.417_beta
EnderIO-1.7.10-2.3.0.418_beta
EnderCore-1.7.10-0.2.0.30_beta

Other mods used to test with:
CoFHCore-[1.7.10]3.0.3-303
ThermalDynamics-[1.7.10]1.1.0-161
ThermalExpansion-[1.7.10]4.0.3B1-218
ThermalFoundation-[1.7.10]1.2.0-102

Setup / steps to reproduction:
http://imgur.com/wNjoqyQ
From left to right:
- creative energy cell (set to max output 2000 rf/t)
- "imput" tesseracks (both set to same frequency 1, left one set to send, right one to receive energy)
- capacitor bank (1x3x3, left side set to pull, right one to push, make sure you have the energy "screens" like in the picture)
- "output" tesseracks (both set to same frequency 2, left one set to send, right one to receive energy)
- resonant energy cell (set to max input 1000 rf/t)
In the picture we have two sides. On the left side of the capacitor bank is input, this side charges it. Right side is output, it sends energy from it into resonant energy cell.

The issue:
Normally, it will show that it is receiving 2000 rf and sending 1000 rf (on the vertical screen which say both in / out; or 1000 rf on the small one). Exit the map and then go back into it. Screen on the battery show 1000 rf in and 2000 rf out. Input has decreased by 50% and output increased 2X. However, battery is still charging and so does the resonant energy cell with no change (is charging with 1000 rf/t, not with 2000 as CapBank screen say) Again, Relog. Battery show different values, again. Again, input decreased by 50% and output increased by 2X. Waila and CapBank gui will also show the same exact incorrect values.

After relogging, CapBank shows that it is outputting 2000 rf/t even if the resonant energy cell is set to only accept 1000 rf/t. Set the input to 0. CapBank is still outputting 1000 rf/t. Disable (remove / provide redstone signal..) Tesserack that is on the right side of the CapBank. It will stop outputting energy.

Other notes:
- this happens even if Tesserack is connected to CapBank via energy cable / conduit
- it seem to happen with TE Tesseracks only
- Transdimensional Receiver doesn't seem to be affected
- it can be fixed by restarting the game

**Comment by HenryLoenwind** [comment]
Can you do us a favor and try your test setup with this debug version? http://loenwind.info/EnderIO-1.7.10-2.3.0.418pre2_beta.jar

It will output information about what the cap bank sends into the logfile. Please check what it says. Thank you.

(This version is compiled with Oracle's JDK, which has a bug. So it needs Immibis Microparts to be installed if you want to place conduits.)

**Comment by artubi** [comment]
I can't upload the whole log because it's too big. But there are few excerpts from it:
Notes:
- same exact setup
- I'm referring to Tesserack that is connected on the right side of the battery that is set to Push, On/Off refers to its settings, whether it's sending on low redstone signal (on) or high redstone signal (off)
- also I'm referring to Resonant Energy Cell and its input settings

**Normal, Tesserack off**
[21:45:09] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 3252/id=0: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@46fe0a73 at x=816 y=56 z=-576. It took 0 RF.
**Normal, Tesserack on, Cell max input 1000 RF**
[21:45:56] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 3439/id=0: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@46fe0a73 at x=816 y=56 z=-576. It took 1000 RF.

**1st relog, Tesserack on, Cell max input 1000 RF**
[21:46:59] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 4181/id=1: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@29098bc7 at x=816 y=56 z=-576. It took 2000 RF.
**1st relog, Tesserack on, Cell max input 0 RF**
[21:47:28] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 4288/id=1: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@29098bc7 at x=816 y=56 z=-576. It took 1000 RF.
**1st relog, Tesserack off**
[21:47:49] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 4390/id=1: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@29098bc7 at x=816 y=56 z=-576. It took 0 RF.

**2nd relog did absolutely nothing different**

**3rd relog, Tesserack on, Cell max input 1000 RF**
[21:49:35] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 5744/id=3: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@474ba46b at x=816 y=56 z=-576. It took 3000 RF.
**3rd relog, Tesserack on, Cell max input 0 RF**
[21:50:11] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 5832/id=3: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@474ba46b at x=816 y=56 z=-576. It took 2000 RF.
**3rd relog, Tesserack off**
[21:51:01] [Server thread/INFO] [STDOUT]: [crazypants.enderio.machine.capbank.network.CapBankNetwork:sendPowerTo:300]: Tick 5949/id=3: Offered 225000 RF to cofh.thermalexpansion.block.ender.TileTesseract@474ba46b at x=816 y=56 z=-576. It took 0 RF.

---
I'm not sure whether it'll help but I did some tests with Power Monitors.
Same exact setup except there are conduits inbetween and Creative Energy Cell on back of the power monitors to avoid tampering with readings. Sides of the bank that are in touch with them are set to disabled.
Setup: http://imgur.com/egZO0L8
I will refer the Power Monitors (from left to right) as  P1, P2, P3, P4.

Results:
**Normal**
P1: 2000 RF/t Out, 2000 RF/t In
P2: 0 RF/t Out, 2000 RF/t In
P3: 1000 RF/t Out, 1000 RF/t In
P4: 1000 RF/t Out, 1000 RF/t In
**1st relog**
P1: 2000 RF/t Out, 2000 RF/t In
P2: 0 RF/t Out, 1000 RF/t In
P3: 2000 RF/t Out, 2000 RF/t In
P4: 1000 RF/t Out, 1000 RF/t In
**2nd relog**
P1: 2000 RF/t Out, 2000 RF/t In
P2: 0 RF/t Out, ~660 RF/t In
P3: 3000 RF/t Out, 3000 RF/t In
P4: 1000 RF/t Out, 1000 RF/t In

1st and 4th Power monitor didn't show any issue. Only 2nd and 3rd did show some inconsistency.
However, the ones that are connected to Tesserack and battery were giving me the same exact numbers of in/outcomming energy as battery.

**Comment by HenryLoenwind** [comment]
Ok, that is helpful. You see the number after "took"? That is what the tesseract actually took. There's absolutely no Ender IO code touching that number. So the tesseract is behaving funny here, the cap bank is doing fine.

(I don't have the code open right now, but the call is something like ```took=otherTileEntity.receiveEnergy(side, offeredAmound)```)

**Comment by artubi** [comment]
Well, I guess you're right. Tesserack is the issue, not the CapBank.
I did some tests with other batteries. Same exact setup as in picture (with conduits and power monitors) I've posted except the in the middle was:
- Mekanism, multiblock Induction Cell
- Mek, Energy Cube
- TE Resonant Energy Cell

After relogging I get same exact result (issue) as above. I should have done these tests before reporting it. My bad.

However, this doesn't happen when the setup is simplified. Remove the middle battery and Tesseracks connected to it (creating Creative Cell > Tesserack > Tesserack > Resonant Cell) and it'll work even after relog. 

I'll report or refer it to team COFH and until then, its either Tesseracks and not relogging or Transdimensional Receiver.
